app/main/views/service_settings.py

```diff
@@ -938,17 +938,26 @@ def set_free_sms_allowance(service_id):
 @main.route("/services/<uuid:service_id>/service-settings/set-message-limit", methods=["GET", "POST"])
 @user_is_platform_admin
 def set_message_limit(service_id):
+    notification_type = request.args.get("notification_type", "").lower() or None
+    if notification_type not in {None, "letter", "email", "sms"}:
+        abort(400)
```

Comment:
Think this is cleaner using the existing [`template_type` URL convertor](https://github.com/alphagov/notifications-admin/blob/main/app/url_converters.py#L11-L13)
```suggestion
@main.route("/services/<uuid:service_id>/service-settings/set-message-limit", methods=["GET", "POST"])
@main.route("/services/<uuid:service_id>/service-settings/set-message-limit/<template_type:notification_type>", methods=["GET", "POST"])
@user_is_platform_admin
def set_message_limit(service_id, notification_type=None):
```